libexec/rtld-elf/rtld_c18n.c

```diff
@@ -761,34 +779,38 @@ tramp_hook_impl(void *rcsp, int event, void *target, const Obj_Entry *obj,
 	const char *sym;
 	const char *callee;
 
+	uintptr_t *stack;
 	compart_id_t caller_id;
 	const char *caller;
 
 	struct utrace_rtld ut;
 	static const char rtld_utrace_sig[RTLD_UTRACE_SIG_SZ] = RTLD_UTRACE_SIG;
 
-	sym_num = def == NULL ? 0 : def->st_name;
-	sym = def == NULL ? "<unknown>" : strtab_value(obj, def->st_name);
-	callee = comparts.data[obj->compart_id]->name;
+	if (ld_compartment_utrace != NULL) {
+		sym_num = def == NULL ? 0 : def->st_name;
+		sym = def == NULL ? "<unknown>" : strtab_value(obj, sym_num);
```

Comment:
Checking def then using sym_num is a little odd. Though so is using sym_num, which is really sym_name_strtab_offset, for the utrace entry, especially when the name's already there...

Reply:
Good catch! I intended to pass the symbol table entry number but somehow ended up passing the string table offset of the symbol name.